### ISSUE TITLE
Morphological segmentation of Turkish words

### ISSUE THREAD

**Issue by andmek** [google-research/turkish-morphology#2]
I need to segment Turkish words. What I need is the morphemes that correspond to the surface form of an orthographic word. How can I use this morphological analyzer for this purpose?

**Comment by alerque** [comment]
I'm presuming you mean "segment" as in break down the root and suffix structure, not as in to break at syllable boundaries right?

**Comment by andmek** [comment]
Yes, it is to break down the root (stem) and affix structure.

**Comment by alerque** [comment]
In that case I'll be watching this issue because I have a use-case for doing this myself!

**Comment by andmek** [comment]
Can the output of the morphophonemic model (the word stem and the meta-morphemes of an orthographic word) be considered as the "morphological segmentation" of the word?

**Comment by ozturel** [comment]
Could you elaborate your use case?

> Can the output of the morphophonemic model (the word stem and the meta-morphemes of an orthographic word) be considered as the "morphological segmentation" of the word?

Short answer: It depends on your use case. The word stem is already present in the output analyses strings. You can use meta-morphemes as a canonical representation of the affix. Though, keep in mind that meta-morphemes could have more than one surface realization given the morphophonemic processes (e.g. vowel harmony).

For an input word (e.g. 'arabaları') you would potentially get many morphological analyses as output, for instance: 
(araba[NN]+[PersonNumber=A3sg]+lArH[Possessive=P3pl]+[Case=Bare])+[Proper=False]
(araba[NN]+[PersonNumber=A3sg]+lArH[Possessive=P3pl]+[Case=Nom])+[Proper=False]
(araba[NN]+lAr[PersonNumber=A3pl]+SH[Possessive=P3sg]+[Case=Bare])+[Proper=False]
(araba[NN]+lAr[PersonNumber=A3pl]+SH[Possessive=P3sg]+[Case=Nom])+[Proper=False]
(araba[NN]+lAr[PersonNumber=A3pl]+[Possessive=Pnon]+YH[Case=Acc])+[Proper=False]
...

For a given word-analysis pair:
_word:_ arabaları 
_analyses:_ (araba[NN]+lAr[PersonNumber=A3pl]+[Possessive=Pnon]+YH[Case=Acc])+[Proper=False]

Do you want to obtain surface form segmentation as below?
araba+lar+ı

**Comment by andmek** [comment]
Yes, I want to obtain surface form segmentation. For example, for the word "arabaları" -- "araba+lar+ı".

**Comment by andmek** [comment]
In your code base, I couldn't get the dedicated morpheme segmentation grammar that marks morpheme boundaries (e.g. evlerinde -- ev+ler+i+nde or arabaları -- araba+lar+ı). Could you please give me any suggestion?

**Comment by ozturel** [comment]
That's right, you can't get the morpheme segmentation of words as you like with the current API, and I don't think there is an easy hack to do it.

Eventually we would like to add this functionality, however we don't have the bandwidth to add it in the immediate/short term.

I'm keeping this Issue open and I will update here when we have some progress on this request.

**Comment by andmek** [comment]
Thanks @ozturel for your reply. I hope you will add this functionality soon. Besides, I appreciate that you have restored the stems of some words which have changed through phonological processes (e.g., 'yürek' for yüreğinizle and 'metin' for metnindeki).
